src/sql/src/plan/query.rs

```diff
@@ -4255,6 +4237,71 @@ fn plan_literal<'a>(l: &'a Value) -> Result<CoercibleScalarExpr, PlanError> {
     Ok(expr.into())
 }
 
+fn validate_window_function_plan<'a>(
+    ecx: &ExprContext,
+    Function {
+        name,
+        args,
+        filter,
+        over,
+        distinct,
+    }: &'a Function<Aug>,
+) -> Result<
+    (
+        &'a WindowSpec<Aug>,
+        Vec<CoercibleScalarExpr>,
+        Vec<HirScalarExpr>,
+    ),
+    PlanError,
+> {
+    if !ecx.allow_windows {
+        sql_bail!("window functions are not allowed in {}", ecx.name);
+    }
+
+    // Various things are duplicated here and below, but done this way to improve
```

Comment:
This comment is talking about duplicated things in `plan_function`. Since we've split that up now, this comment could be removed or changed to say maybe:

> Various things are duplicated here and in `plan_function` to improve error messages.

Reply:
Nice catch, thanks!